### DISCUSSION TITLE
Cachebench memory behavior upon cache hits

### DISCUSSION THREAD

**Discussion by kevins981** [facebook/CacheLib#209] [Q&A] [answered]
Hi CacheLib developers,

In the cachebench stressor, upon a [get operation cache hit](https://github.com/facebook/CacheLib/blob/67d2d8be64206b6c75e123483c3ebe202b548a1c/cachelib/cachebench/runner/CacheStressor.h#L369), it seems like the [returned item handle](https://github.com/facebook/CacheLib/blob/67d2d8be64206b6c75e123483c3ebe202b548a1c/cachelib/cachebench/runner/CacheStressor.h#L354) is not used by cachebench. My understanding is that (please correct me if I am wrong) since the item handle is not used, the item value is actually not accessed. I imagine that in a real CacheLib use case, the item value has to be accessed at least once (e.g. sent throughput the network). From a memory behavior perspective, does this means that the current cachebench is underestimating the number of memory accesses? 

If my understanding above is correct, what do you think would be a reasonable way to access the item values? Would a string copy/memcpy work?

Note that I think that the item keys are already being accessed within `cache_->find()`. But since in many cases the item values can be much larger than keys, the question remains.

Thank you for your time,
Kevin

**Comment by jaesoo-fb** [comment, marked as answer]
Hi @kevins981,

Actually, the Cachebench supports accessing all bytes of found item. You can enable the option using ["touchValue"](https://github.com/facebook/CacheLib/blob/67d2d8be64206b6c75e123483c3ebe202b548a1c/cachelib/cachebench/util/Config.h#L211)

Also, please note that the main use of cachebench is not to simulate the application behavior.

**Reply by kevins981** [reply]
Thank you for the prompt solution!
